torchci/lib/bot/retryBot.ts

```diff
@@ -0,0 +1,76 @@
+import { Probot } from "probot";
+
+const FAILURE_CONCLUSIONS = ["failure", "cancelled", "timed_out"];
+
+function retryBot(app: Probot): void {
+  app.on("workflow_run.completed", async (ctx) => {
+    const workflowName = ctx.payload.workflow_run.name;
+    const attemptNumber = ctx.payload.workflow_run.run_attempt;
+    if (
+      ctx.payload.workflow_run.conclusion === "success" ||
+      ctx.payload.workflow_run.head_branch !== "master" ||
+      attemptNumber > 1 ||
+      !["lint", "pull"].includes(workflowName)  // only do lint and pull for now because they are fast
```

Comment:
When this is proven to be effective, I hope we can have this in trunk as well, i.e. GitHub infra flakiness in trunk https://hud.pytorch.org/pytorch/pytorch/commit/2288a1c8065dd1d43410089719028087bd40e997